label_sync/labels.yaml

```diff
@@ -257,12 +257,6 @@ default:
       target: both
       prowPlugin: label
       addedBy: anyone
-    - color: c7def8
-      description: Categorizes issue or PR as related to design.
-      name: kind/design
-      target: both
-      prowPlugin: label
-      addedBy: anyone
```

Comment:
This will prevent the label from being added to new repos, but will not actually delete the label from existing repos.  You need to use `deleteAfter` for that.  

But, doing so will delete the label from any previously closed issues that use it, and we'll lose that historical context if they have no other kind/ labels. e.g. the query [`is:closed repo:kubernetes/kubernetes label:kind/design -label:kind/feature`][1] yields 798 results

Without looking deeply at the code I'm not sure what the behavior of label syncer is when asked to simultaneously:
1. delete an org-wide label, add as repo-specific label
2. make an org-wide label a `previously` for another org-wide label, but preserve it for specific repos

If we care about preserving historical categorization, I foresee:
- testing what label_sync would do with case 2 (if there aren't already unit tests that prove this out)
- if it can't gracefully handle case 2, manually adding kind/feature to kind/design issues that don't already have it (closed or not)
- testing what label_sync would do with case 1
- if it can't gracefully handle case 1, manually delete the label from all repos that aren't specifically using kind/design

[1]: https://github.com/issues?q=is%3Aclosed+repo%3Akubernetes%2Fkubernetes+label%3Akind%2Fdesign+-label%3Akind%2Ffeature

Reply:
Thanks @spiffxp, this is really helpful!

A few clarifications:
> This will prevent the label from being added to new repos, but will not actually delete the label from existing repos.

Yes, that is the intention. From https://github.com/kubernetes/community/issues/5641, the general consensus was that `kind/design` be removed from `k/k` alone and not all repositories. If new repositories would like the `kind/design` label, then they would need to opt-in by adding their repositories to `labels.yaml`: https://github.com/kubernetes/community/issues/5641#issuecomment-882506726. This change of opting in will be communicated through the k-dev mailing list. Relevant slack discussion [here](https://kubernetes.slack.com/archives/C1TU9EB9S/p1626697872447500).

> make an org-wide label a previously for another org-wide label, but preserve it for specific repos

I had tried this, `label_sync` unfortunately does not allow this.

> if it can't gracefully handle case 2, manually adding kind/feature to kind/design issues that don't already have it (closed or not)

ack
I've done a sweep for open issues/PRs in `k/k` but not the closed ones. 

> testing what label_sync would do with case 1

From my initial understanding, this should work fine. But I will take closer look and try adding a few tests to be sure.

Reply:
Hey @spiffxp! Quick update - I ran a few tests by running `label_sync` against a dummy repo:

> delete an org-wide label, add as repo-specific label

The behavior of doing this is as expected, that is:
- When a new repo is added, it will not get this deleted label, unless it is explicitly added in the `repos` section
- Repos that already have the deleted label, will continue to have it

> make an org-wide label a `previously` for another org-wide label, but preserve it for specific repos

If something like this is done, then it will sync at an org level, and all existing `kind/design` labels will be replaced with `kind/feature` (even in repos that actually use `kind/design`)
```yaml
---
default:
  labels:
    - color: c7def8
      name: kind/feature
      description: Feature
      previously:
      - color: c7def8
        name: kind/design
        description: Design
repos:
  org/repo:
    labels:
      - color: c7def8
        name: kind/design
        description: Design
```